pkg/reconciler/pipelinerun/resources/pipelineref.go

```diff
@@ -149,41 +124,48 @@ func (l *LocalPipelineRefResolver) GetPipeline(ctx context.Context, name string)
 }
 
 // resolvePipeline accepts an impl of remote.Resolver and attempts to
-// fetch a pipeline with given name. An error is returned if the
-// resolution doesn't work or the returned data isn't a valid
-// *v1beta1.Pipeline.
-func resolvePipeline(ctx context.Context, resolver remote.Resolver, name string) (*v1beta1.Pipeline, *v1beta1.RefSource, error) {
+// fetch a pipeline with given name and verify the v1beta1 pipeline if trusted resources is enabled.
+// An error is returned if the resolution doesn't work, the verification fails
+// or the returned data isn't a valid *v1beta1.Pipeline.
+func resolvePipeline(ctx context.Context, resolver remote.Resolver, name string, k8s kubernetes.Interface, verificationPolicies []*v1alpha1.VerificationPolicy) (*v1beta1.Pipeline, *v1beta1.RefSource, error) {
 	obj, refSource, err := resolver.Get(ctx, "pipeline", name)
 	if err != nil {
 		return nil, nil, err
 	}
-	pipelineObj, err := readRuntimeObjectAsPipeline(ctx, obj)
+	pipelineObj, err := readRuntimeObjectAsPipeline(ctx, obj, k8s, refSource, verificationPolicies)
 	if err != nil {
-		return nil, nil, fmt.Errorf("failed to convert obj %s into Pipeline", obj.GetObjectKind().GroupVersionKind().String())
+		return nil, nil, err
 	}
 	return pipelineObj, refSource, nil
 }
 
 // readRuntimeObjectAsPipeline tries to convert a generic runtime.Object
 // into a *v1beta1.Pipeline type so that its meta and spec fields
 // can be read. v1 object will be converted to v1beta1 and returned.
+// v1beta1 Pipeline will be verified if trusted resources is enabled
 // An error is returned if the given object is not a
-// PipelineObject or if there is an error validating or upgrading an
+// PipelineObject, trusted resources verification fails or if there is an error validating or upgrading an
 // older PipelineObject into its v1beta1 equivalent.
 // TODO(#5541): convert v1beta1 obj to v1 once we use v1 as the stored version
-func readRuntimeObjectAsPipeline(ctx context.Context, obj runtime.Object) (*v1beta1.Pipeline, error) {
+func readRuntimeObjectAsPipeline(ctx context.Context, obj runtime.Object, k8s kubernetes.Interface, refSource *v1beta1.RefSource, verificationPolicies []*v1alpha1.VerificationPolicy) (*v1beta1.Pipeline, error) {
 	switch obj := obj.(type) {
 	case *v1beta1.Pipeline:
+		// Verify the Pipeline once we fetch from the remote resolution, mutating, validation and conversion of the resource should happen after the verification.
```

Comment:
```suggestion
		// Verify the Pipeline once we fetch from the remote resolution, mutating, validation and conversion of the resource should happen after the verification, since signatures are based on the remote pipeline contents
```